lib/media/drm_engine.js

```diff
@@ -774,6 +783,48 @@ shaka.media.DrmEngine.prototype.configureClearKey_ = function() {
 };
 
 
+/**
+ * Returns the DrmInfo that is generated by drm configation.
+ * It activates DrmEngine if drm configs have keySystems.
+ * @param {!shakaExtern.Manifest} manifest
+ * @return {(Array.<{object}>|null)}
+ * @private
+ */
+shaka.media.DrmEngine.prototype.getDrmInfosByConfig_ =
+    function(manifest) {
+  var config = this.config_;
+  if (!config) {
+    return null;
+  }
+
+  var serverKeys = Object.keys(config.servers || {});
+
+  if (!serverKeys.length) {
+    return null;
+  }
+
+  var isEncryptedContent = manifest.periods.every(function(period) {
+    period.variants.every(function(variant) {
```

Comment:
Same here.  I believe this should be `some()`, as well.  Also, you forgot a `return` here, so the outermost `every()` function has no return value.

Reply:
Thank you for reviewing, @joeyparrish . Last min I reversed this behavior from unencrypted to encrypted. I should have noticed this mistake. Apologies. On the other hand, it seems @TheModMaker 's suggestion more simple than mine, but it throws an error. I was going to trace that error, but I was interrupted by something else and I could not spend enough time. Do you think it is not necessary to chase his solution?  Thanks again.

Reply:
It is simpler, but I think we should keep ignoreDrmInfo separate.  It serves a different purpose, and it isn't on by default.  What you are changing here should, in my opinion, be default behavior to match developer expectations.